frontend/src/screens/MapScreen.js

```diff
@@ -0,0 +1,124 @@
+  const placeRef = useRef(null);
+  const markerRef = useRef(null);
+
+  useEffect(() => {
+    const fetch = async () => {
+      const { data } = await Axios('/api/config/google');
+      setGoogleApiKey(data);
+      getUserCurrentLocation();
+    };
+    fetch();
+  }, []);
+
+  const onLoad = (map) => {
+    mapRef.current = map;
+  };
+
+  const onMarkerLoad = (marker) => {
+    markerRef.current = marker;
+  };
+  const onLoadPlaces = (place) => {
+    placeRef.current = place;
+  };
+  const onIdle = () => {
+    setLocation({
+      lat: mapRef.current.center.lat(),
+      lng: mapRef.current.center.lng(),
+    });
+  };
+  const onPlacesChanged = () => {
+    const place = placeRef.current.getPlaces()[0].geometry.location;
+    setCenter({ lat: place.lat(), lng: place.lng() });
+    setLocation({ lat: place.lat(), lng: place.lng() });
+  };
+  const dispatch = useDispatch();
+  const onConfirm = () => {
+    const places = placeRef.current.getPlaces();
+    if (places && places.length === 1) {
+      // dispatch select action
+      dispatch({
+        type: USER_ADDRESS_MAP_CONFIRM,
+        payload: {
+          lat: location.lat,
+          lng: location.lng,
+          address: places[0].formatted_address,
+          name: places[0].name,
+          vicinity: places[0].vicinity,
+          googleAddressId: places[0].id,
+        },
+      });
+      alert('location selected successfully.');
+      props.history.push('/shipping');
+    } else {
+      alert('Please enter your address');
+    }
+  };
+
+  const getUserCurrentLocation = () => {
+    if (!navigator.geolocation) {
+      alert('Geolocation os not supported by this browser');
+    } else {
+      navigator.geolocation.getCurrentPosition((position) => {
+        setCenter({
+          lat: position.coords.latitude,
+          lng: position.coords.longitude,
+        });
+        setLocation({
+          lat: position.coords.latitude,
+          lng: position.coords.longitude,
+        });
+      });
+    }
+  };
+
+  return googleApiKey ? (
+    <div className="full-container">
+      <LoadScript libraries={libs} googleMapsApiKey={googleApiKey}>
+        <GoogleMap
+          id="smaple-map"
+          mapContainerStyle={{ height: '100%', width: '100%' }}
+          center={center}
+          zoom={15}
+          onLoad={onLoad}
+          onIdle={onIdle}
+        >
+          <StandaloneSearchBox
+            onLoad={onLoadPlaces}
+            onPlacesChanged={onPlacesChanged}
+          >
+            <div className="map-input-box">
+              <input type="text" placeholder="Enter your address"></input>
+              <button type="button" className="primary" onClick={onConfirm}>
+                Confirm
+              </button>
+            </div>
+          </StandaloneSearchBox>
+          <Marker position={location} onLoad={onMarkerLoad}></Marker>
+        </GoogleMap>
+      </LoadScript>
+    </div>
+  ) : (
+    <LoadingBox></LoadingBox>
+  );
+}
```

Comment:
In the second 25:28 you press the OK in window.confirm()... but when you go back to ShippingAddressScreen in the next second 25:29 you haven´t got the address fields filled ... you select manually the address from your Google Account ... 
Something like this happends at first in 0:22 when you are going to do the test and you already have the fields filled ...
It is ok ? or maybe what have to happend is, when you choose your address in GoogleMaps and you press OK in the window.confirm(), the fields would have to fill them automatically ... anyone more realized that and think it ??